### PR TITLE
Add option to fetch API key from AWS SSM

### PR DESCRIPTION
**Pull request by demolitionmode**
### What does this PR do?

- Adds option to fetch API Key from AWS SSM.
- Updates Parameters description to show all the options

### Motivation

I use the parameter store for all other things, I like to be consistent where possible.

### Additional Notes

Related issue: https://github.com/DataDog/datadog-serverless-functions/issues/195

### CODE REVIEW

File: aws/logs_monitoring/lambda_function.py [hunk 1]
```diff
@@ -220,6 +237,12 @@ def compileRegex(rule, pattern):
     DD_API_KEY = boto3.client("secretsmanager").get_secret_value(
         SecretId=SECRET_ARN
     )["SecretString"]
+elif "DD_API_KEY_SSM_NAME" in os.environ:
```
Comment (DarcyRaynerDD) [inline aws/logs_monitoring/lambda_function.py:240]: Could we rename this it DD_SSM_API_KEY, just to keep it consistent with DD_KMS_API_KEY

**Comment by DarcyRaynerDD** [comment]
Hi marshall0705 👋 
Thanks for the PR, this looks super useful. I left one small nitpick, but otherwise we should be good to add this to the next release.

**Comment by demolitionmode** [comment]
Hi @DarcyRaynerDD 
That's good to hear. I welcome all comments, even nitpicking 😄, however I don't seem to be able to see your comment.

**Review by DarcyRaynerDD — CHANGES_REQUESTED** [review]

File: aws/logs_monitoring/lambda_function.py (hunk 1 above)
Comment (DarcyRaynerDD) [reply aws/logs_monitoring/lambda_function.py:240]: @marshall0705, I forgot to submit the review, doh

Comment (demolitionmode) [reply aws/logs_monitoring/lambda_function.py:240]: I had considered naming it that. It was a choice between the convention used for the secrets manager var, and the convention used for the KMS var.
I chose this name due to DD_KMS_API_KEY and DD_API_KEY both actually containing the key (albeit one of them encrypted), whereas DD_API_KEY_SECRET_ARN is a reference to a remote location, as is DD_API_KEY_SSM_NAME.
If you would still like the name changing, I'm fine to do so.

Comment (DarcyRaynerDD) [reply aws/logs_monitoring/lambda_function.py:240]: That makes sense, I think we can keep your naming convention then. I'll merge this and roll it out into the next release.

**Review by DarcyRaynerDD — APPROVED** [review]

**Pull request merged**